internal/services/logz/logz_monitor_resource.go

```diff
@@ -0,0 +1,406 @@
+		Delete: resourceLogzMonitorDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.LogzMonitorID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[\w\-]{1,32}$`),
+					`The name length must be from 1 to 32 characters. The name can only contain letters, numbers, hyphens and underscore.`,
+				),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"company_name": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"enterprise_app_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"single_sign_on_url": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"logz_organization_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"plan_data": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"billing_cycle": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"MONTHLY",
+								"WEEKLY",
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"effective_date": {
+							Type:             pluginsdk.TypeString,
+							Required:         true,
+							ForceNew:         true,
+							DiffSuppressFunc: suppress.RFC3339Time,
+							ValidateFunc:     validation.IsRFC3339Time,
+						},
+
+						"plan_id": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"100gb14days",
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"usage_type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"PAYG",
+								"COMMITTED",
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+					},
+				},
+			},
+
+			"user_info": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"email_address": {
```

Comment:
is address redundant?
```suggestion
						"email": {
```